src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -168,6 +168,31 @@ enum AnchorFailure {
     RustdocAnchorConflict(Res),
 }
 
+#[derive(Clone, Debug, Hash, PartialEq, Eq)]
+struct CacheKey {
+    module_id: DefId,
+    dis: Option<Disambiguator>,
+    path_str: String,
+    extra_fragment: Option<String>,
+}
+
+impl CacheKey {
+    fn new(
+        module_id: DefId,
+        dis: Option<Disambiguator>,
+        path_str: String,
+        extra_fragment: Option<String>,
+    ) -> Self {
+        Self { module_id, dis, path_str, extra_fragment }
+    }
+}
```

Comment:
nit: for trivial functions like this I think it makes sense to just use the `Self { ... }` constructor directly.

Reply:
Yeah, it still feels weird that I can do that even if I intended the fields to be somewhat private. 😅

Reply:
If you like you can move it into a separate module with a privacy boundary, but for something this simple I don't think it's necessary. I do want to split this file up at some point though.